### ISSUE TITLE
eof-newline options

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#135]
Can I adapt this rule to accept `"always"|"never"`? At the moment it doesn't accept any options, which is usually reserved for `-no-` rules. Also, I think the "never" option will be useful e.g. the suitcss styleguide specifies "Strip all end-of-line and end-of-file whitespace".

**Comment by davidtheclark** [comment]
Yes, that sounds good to me.

**Comment by MoOx** [comment]
Be careful about this recommendation.
I think most editor are bad at displaying \n.

This input
````
1\n
2\n
```

will be displayed in most editor like

````
1
2

```

except in vim (because this one consider \n to make a jump only when there is another character after)

The interesting part is to make sure you got an \n at the end of each line, mainly for diff (to avoid changing the last line cause you added a \n for example).

I think suitcss is recommanding not having multiple empty lines at the end like this

```
1
2


```

So not sure it's interesting to enforce having no \n at all since it's not a good practice.

**Comment by jeddy3** [comment]
> I think suitcss is recommanding not having multiple empty lines at the end like this

That makes more sense :) I hadn't thought about it in that way.

Should we have a rule to enforce that e.g. `no-multiple-empty-lines`? For example, a user can combine the rules:

```js
eof-newline: 2,
no-multiple-empty-lines: 2,
```

To enforce something like "File must end with newline, but not multiple newlines". The `no-multiple-empty-lines` rule is generic enough to apply everywhere too.

Also, are we sure we want to no longer reserve rules without options for `-no-` rules? I guess it would give us a little more flexibility when naming rules, but it would be slightly less consistent (I think the current approach is easier to reason and document, which I think will help keep the rules consistent as new ones are added by other contributors). If we want to keep with the current approach, we could rename the rule to something like `no-eof-missing-newline`:

```js
no-eof-missing-newline: 2,
no-multiple-empty-lines: 2,
no-eol-whitespace: 2
```

Finally, should we slightly renaming `eol-no-whitespace` to `no-eol-whitespace`, as I've done above, to keep it consistent with the other rules that don't apply to an identifiable *thing* in the css?

**Comment by davidtheclark** [comment]
> Should we have a rule to enforce that e.g. no-multiple-empty-lines

Yes, I think.

> no-eof-missing-newline

Seems reasonable to me.

The whole convention with `no-*` rules seems reasonable. Really we are stopping a mistake in every case.
